umh-core/README.md

```diff
@@ -1,178 +1,219 @@
+| Health probes            | `/live` & `/ready` pinged every 1 s; logs scanned for fatal patterns; metrics watched for `disk_full`, partition alerts                | Any issue shows a yellow **Redpanda** badge in the console                                    |
+
+---
+
+## 6  Startup timeline (from code)
+
+```
+Stopped ──▶ Starting
+            • S6 service is up
+            • Wait 5 s uptime to deem “config loaded”
+            • /ping & /ready healthy
+            • No critical logs/metrics for 10 s
+            ──▶ Idle      (≤ 15 s total or retries kick in)
+Idle ──▶ Active as soon as throughput rises
 ```
 
-**Volume Mounts:**
-- `/data/umh-lite-bootstrap.yaml` is the read-only bootstrap configuration.
-- `umh-lite-state` is a persistent volume used for stateful data (Redpanda offsets, pipeline state, etc.).
+---
+
+## 7  Sizing guide
 
-### 3. How Desired State Updates Work
+**Start with → 2 vCPU · 4 GB RAM · 40 GB SSD**
 
-**umh-core** is designed to be dynamic:
-- **External Updates via CI/CD:**  
-  Users can update the YAML configuration in source control and deploy new versions via a CI/CD pipeline. When the new YAML is mounted, the agent detects the updated desired state and adjusts the pipelines accordingly.
-- **Internal Updates via the Agent:**  
-  The agent continuously communicates with the UMH Management Console. Change requests (such as new protocol converters or bridge adjustments) are written into an updated YAML file. The agent reads this file and transitions its internal state (e.g., from `StateRunning` to `StateConfigChanged`, then to `StateChangingConfig`, and finally to `StateStarting` before returning to `StateRunning`).
+### What that box handles
 
-In either case, the updated YAML drives the agent’s finite state machine, ensuring that changes are applied deterministically.
+* **≈ 9 protocol-converter DFCs** (e.g. OPC UA ➜ Redpanda) **plus one bridge DFC** that forwards from the local Redpanda to an external MQTT broker
+* **≈ 900 tags at 1 message / second each**
+* Keeps **seven days** of history under the default cluster retention (`log_retention_ms = 7 days`)
+* Runs comfortably below 70 % CPU and I/O on a Hetzner CAX21 / CX32-class VM or Raspberry Pi 4
 
-### 4. Metrics
+### Disk usage in practice
 
-The container exposes a Prometheus metrics endpoint at port `4195`:
-- **Metrics Endpoint:** `http://<device-ip>:4195/metrics`  
-  This endpoint provides basic internal metrics from the agent, Benthos pipelines, and Redpanda.
-- While metrics are supported, the primary focus is on data flow reliability and dynamic configuration, not on extensive local metrics visualization.
+Redpanda writes **1 GiB segments**; a segment can be deleted only after it is closed.
+With Snappy compression, a typical 200 B JSON payload shrinks to ≈ 20 B.
```

Comment:
This must either be set by the producer or via rpk.
See also: https://docs.redpanda.com/current/reference/properties/topic-properties/#compressiontype & https://www.confluent.io/blog/apache-kafka-message-compression/